File: hw/yanez/u3/HW16MazeGenerator/model/Maze.py
from model.Room import Room
from model.Direction import Direction
from controller.RandomMazeGenerator import RandomMazeGenerator
from controller.MazePrinter import MazePrinter

class Maze:
    def __init__(self, rows: int, cols: int):
        self.rooms = [[None for _ in range(cols)] for _ in range(rows)]
        self.initialize_rooms()
        self.generate_maze()

    def initialize_rooms(self):
        for row in range(len(self.rooms)):
            for col in range(len(self.rooms[0])):
                self.rooms[row][col] = Room(row, col)

    def generate_maze(self):
        generator = RandomMazeGenerator()
        generator.generate(self.rooms)
        self.rooms[0][0].get_wall(Direction.WEST).open()  # Entrada
        self.rooms[-1][-1].get_wall(Direction.EAST).open()  # Salida

    def display(self):
        printer = MazePrinter(self.rooms)
        printer.print()

File: hw/unda/U3/HW16MazeGenerator/HW16MazeGenerator/src/main/java/ec/edu/espe/hw16mazegenerator/model/Maze.java
package ec.edu.espe.hw16mazegenerator.model;

import ec.edu.espe.hw16mazegenerator.controller.MazeGenerator;
import ec.edu.espe.hw16mazegenerator.controller.MazePrinter;
import ec.edu.espe.hw16mazegenerator.controller.RandomMazeGenerator;

/**
 *
 * @author Mateo Unda
 */
public class Maze {

    private final Room[][] rooms;

    public Maze(int rows, int cols) {
        rooms = new Room[rows][cols];
        initializeRooms();
        generateMaze();
    }

    private void initializeRooms() {
        for (int row = 0; row < rooms.length; row++) {
            for (int col = 0; col < rooms[0].length; col++) {
                rooms[row][col] = new Room(row, col);
            }
        }
    }

    private void generateMaze() {
        MazeGenerator generator = new RandomMazeGenerator();
        generator.generate(rooms);

        // Abrir entrada y salida
        rooms[0][0].getWall(Direction.WEST).open(); // Entrada
        rooms[rooms.length - 1][rooms[0].length - 1].getWall(Direction.EAST).open(); // Salida
    }

    public void display() {
        MazePrinter printer = new MazePrinter(rooms);
        printer.print();
    }
}
